### ISSUE TITLE
Error when compiling with Clang on OSX

### ISSUE THREAD

**Issue by gjasny** [nanoant/CMakePCHCompiler#3]
Hello,

When compiling the example (08a33206c336c22fe3e39027643c737f15a6cc46) I get the following error:

```
$ make VERBOSE=1
/usr/local/Cellar/cmake/3.1.3/bin/cmake -H/Users/gjasny/src/CMakePCHCompiler/example -B/Users/gjasny/src/CMakePCHCompiler/example --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/Cellar/cmake/3.1.3/bin/cmake -E cmake_progress_start /Users/gjasny/src/CMakePCHCompiler/example/CMakeFiles /Users/gjasny/src/CMakePCHCompiler/example/CMakeFiles/progress.marks
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/engine.pch.dir/build.make CMakeFiles/engine.pch.dir/depend
cd /Users/gjasny/src/CMakePCHCompiler/example && /usr/local/Cellar/cmake/3.1.3/bin/cmake -E cmake_depends "Unix Makefiles" /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example/CMakeFiles/engine.pch.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/engine.pch.dir/build.make CMakeFiles/engine.pch.dir/build
make[2]: Nothing to be done for `CMakeFiles/engine.pch.dir/build'.
/usr/local/Cellar/cmake/3.1.3/bin/cmake -E cmake_progress_report /Users/gjasny/src/CMakePCHCompiler/example/CMakeFiles  5
[ 20%] Built target engine.pch
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/engine.dir/build.make CMakeFiles/engine.dir/depend
cd /Users/gjasny/src/CMakePCHCompiler/example && /usr/local/Cellar/cmake/3.1.3/bin/cmake -E cmake_depends "Unix Makefiles" /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example /Users/gjasny/src/CMakePCHCompiler/example/CMakeFiles/engine.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/engine.dir/build.make CMakeFiles/engine.dir/build
Linking CXX shared library libengine.dylib
/usr/local/Cellar/cmake/3.1.3/bin/cmake -E cmake_link_script CMakeFiles/engine.dir/link.txt --verbose=1
/usr/bin/c++   -dynamiclib -Wl,-headerpad_max_install_names  -o libengine.dylib -install_name /Users/gjasny/src/CMakePCHCompiler/example/libengine.dylib CMakeFiles/engine.dir/src/engine.cpp.o CMakeFiles/engine.dir/src/library.cpp.o CMakeFiles/engine.dir/src/prefix.h.pch 
error: invalid value 'precompiled-header' in '-x precompiled-header'
make[2]: *** [libengine.dylib] Error 1
make[1]: *** [CMakeFiles/engine.dir/all] Error 2
make: *** [all] Error 2

```

I'm using CMake 3.1.3 with Xcode 6.1.1 on OSX Yosemite.

Thanks,
Gregor

**Comment by nanoant** [comment]
My fault, sorry, `src/prefix.h` should not be added as a source file to the target. Fixed in deeeab096a6af620668ef3e81f3b2f68eec4260d.

**Comment by mandalin** [comment]
I'm having this same error in my code, can you help me understand the cause ?
